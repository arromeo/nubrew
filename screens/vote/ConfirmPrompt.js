import React from 'react';
import { StyleSheet, View, Text, Image, TouchableOpacity} from 'react-native';
import { Ionicons } from '@expo/vector-icons';

export default class ConfirmPrompt extends React.Component {

  render() {
    return (
      <View style={styles.container}>
        {this.props.couldNotFind &&
          <View style={styles.verticalContainer}>
            <Text style={styles.headerFont}>We couldn't find your drink!</Text>
            <Ionicons name="md-sad" size={50} color="red"/>
            <Text style={styles.paragraphFont}>Sometimes the labels change or maybe we don't have it in our database yet! Try again and reposition the beer label to fill the guide box or searching by the drink's name!</Text>
          </View>
        }
        {!this.props.couldNotFind &&
          <View style={styles.verticalContainer}>
            <Text style={styles.headerFont}>{this.props.data.brewery_name}'s {this.props.data.beer_name}</Text>
            <Image source={{uri: this.props.data.img_url}} style={{height: 200, width: 150}}/>
            <Text>Is this the correct drink?</Text>
          </View>
        }
        {this.props.couldNotFind &&
          <View style={styles.contentContainer}>
            <TouchableOpacity 
              style={styles.buttonStyle}
              onPress={() => {
                this.props.resetToCamera()
              }}>
              <Ionicons name="md-camera" size={25} color="white"/>
            </TouchableOpacity>
            <TouchableOpacity 
              style={styles.buttonStyle}
              onPress={() => {
                this.props.navigate('Find');
              }}>
              <Ionicons name="md-search" size={25} color="red"/>
            </TouchableOpacity>
          </View>
        }
        {!this.props.couldNotFind &&
          <View style={styles.contentContainer}>
            <TouchableOpacity
              style={styles.buttonStyle}
              onPress={() => {
                this.props.navigate('Detail', {
                  category: "Beer",
                  id: this.props.data.beer_id,
                });
              }}>
              <Ionicons name="md-checkmark-circle-outline" size={25} color="green"/>
            </TouchableOpacity>
            <TouchableOpacity 
              style={styles.buttonStyle}
              onPress={() => {
                this.props.resetToCamera()
              }}>
              <Ionicons name="md-camera" size={25} color="white"/>
            </TouchableOpacity>
            <TouchableOpacity 
              style={styles.buttonStyle}
              onPress={() => {
                this.props.navigate('Find');
              }}>
              <Ionicons name="md-close-circle-outline" size={25} color="red"/>
            </TouchableOpacity>
          </View>
        }
      </View>
    )
  }
   
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 15,
    backgroundColor: '#fff',
  },
  headerFont: {
    color: 'black',
    fontWeight: 'bold',
    fontSize: 20,
    textAlign: 'center',
  },
  paragraphFont: {
    color: 'black',
    fontWeight: 'bold',
    fontSize: 18,
    paddingTop: 10,
    width: '90%',
    textAlign: 'center',
  },
  verticalContainer: {
    flex: 1,
    margin: 10,
    justifyContent: "space-around",
    alignItems: 'center',
  },
  contentContainer: {
    flex: 0.5,
    paddingBottom: 5,
    flexDirection: "row",
    justifyContent: "space-around",
    alignItems: 'center',
  },
  buttonStyle: {
    marginTop: 10,
    marginBottom: 10,
    padding: 5,
    flex: 0.2,
    width: '75%',
    borderRadius: 25,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#61170E',
  }
});
