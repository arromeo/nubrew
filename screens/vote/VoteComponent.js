import React from 'react';
import { StyleSheet, View, TouchableOpacity, Text, Slider } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

export default class VoteComponent extends React.Component {
  render() {
    return (
      <View>
        {this.props.voteCast === 'None' &&
          <View style={styles.voteContainer}>
            <TouchableOpacity
              style={styles.buttonStyle}
              onPress={() => {
                this.props.profileUpdate();
                this.props.updateVote(-1, this.props.navigationParams.id, this.props.user_id)
              }}>
              <Ionicons style={{zIndex: 2}} name="md-thumbs-down" size={45} color="red"/>
            </TouchableOpacity>
            <TouchableOpacity
              style={styles.buttonStyle}
              onPress={() => {
                this.props.profileUpdate();
                this.props.updateVote(1, this.props.navigationParams.id, this.props.user_id)
              }}>
              <Ionicons style={{zIndex: 2}} name="md-thumbs-up" size={45} color="green"/>
            </TouchableOpacity>
          </View>
        }
        {this.props.voteCast === 'Liked' &&
          <View style={styles.confirmationContainer}>
            <Ionicons name="md-thumbs-up" size={80} color="green"/>
          </View>
        }
        {this.props.voteCast === 'Disliked' &&
          <View style={styles.confirmationContainer}>
            <Ionicons name="md-thumbs-down" size={80} color="red"/>
          </View>
        }
      </View>
    )
  }
}
const styles = StyleSheet.create({
  voteContainer: {
    flex: 0.5,
    paddingTop: 20,
    paddingBottom: 20,
    paddingLeft: 60,
    paddingRight: 60,
    flexDirection: "row",
    justifyContent: "space-around",
    alignItems: 'center',
  },
  confirmationContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttonStyle: {
    flex: 0.2,
    padding: 10,
    borderRadius: 25,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'lightgrey',
    opacity: 0.8,
  },
});
