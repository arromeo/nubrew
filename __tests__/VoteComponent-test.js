import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import '../__mock__/xhr-mock';

import ConfirmPrompt from '../screens/vote/ConfirmPrompt';
import VoteComponent from '../screens/vote/VoteComponent';

import props from '../__mock__/props';

describe('Vote Components render properly', () => {
  it('Confirm prompt component renders', () => {
    const tree = renderer.create(<ConfirmPrompt navigate={props.navigation.navigate} data={props.data} couldNotFind={props.couldNotFind}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('VoteComponent renders properly', () => {
    const tree = renderer.create(<VoteComponent navigationParams={props.navigationParams}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })
})