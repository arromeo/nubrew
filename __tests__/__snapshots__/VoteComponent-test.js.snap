// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vote Components render properly Confirm prompt component renders 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingTop": 15,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "margin": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "flex": 1,
          "fontSize": 20,
          "fontWeight": "bold",
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      's 
      
    </Text>
    <Image
      source={
        Object {
          "uri": "",
        }
      }
      style={
        Object {
          "height": 200,
          "width": 150,
        }
      }
    />
    <Text>
      Is this the correct drink?
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "paddingBottom": 5,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#61170E",
          "borderRadius": 25,
          "flex": 0.2,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 10,
          "opacity": 1,
          "padding": 5,
          "width": "75%",
        }
      }
    >
      <Text />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#61170E",
          "borderRadius": 25,
          "flex": 0.2,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 10,
          "opacity": 1,
          "padding": 5,
          "width": "75%",
        }
      }
    >
      <Text />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#61170E",
          "borderRadius": 25,
          "flex": 0.2,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 10,
          "opacity": 1,
          "padding": 5,
          "width": "75%",
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;

exports[`Vote Components render properly VoteComponent renders properly 1`] = `<View />`;
